Installation.md

```diff
@@ -0,0 +1,267 @@
+
+```
+$ sudo -s
+$ su - postgres
+$ psql
+    CREATE USER tmc WITH SUPERUSER PASSWORD 'tmc';
+```
+
+#### Set PostgreSQL to authenticate with md5 instead of local ####
+
+Locate the **pg_hba.conf** file with `locate pg_hba.conf`, the file should be located */etc/postgresql/<version number>/main/pg_hba.conf*. Open the file with your preferred text-editor, you need root privileges to edit the file. Around line 90 change
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     peer
+```
+to
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     md5
+```
+
+### TMC-server installation ###
+#### Clone the TMC repository ####
+
+```
+$ cd tmc-server
+$ git clone https://github.com/testmycode/tmc-server.git
+```
+:exclamation: If you are not using a github account, replace the repository submodule URLs with a HTTPS URL in .git/config eg. https://github.com/testmycode/tmc-checkstyle-runner.git
+```
+$ git submodule update --init --recursive
+$ gem install bundler && bundle install
+$ cp config/site.defaults.yml confg/site.yml
+$ env RAILS_ENV=production rake db:reset
+$ env RAILS_ENV=production rake assets:precompile
+```
+
+#### Build sandbox ####
+
+```
+$ cd ext/tmc-sandbox
+$ sudo make
+```
+
+#### Install tmc-sandbox web dependencies ####
+
+```
+$ cd web
+$ bundle install
+$ rake ext
+$ rvmsudo rake test
+```
+
+#### Compile rest of the externals ####
+
+```
+$ cd ../..
+$ rake compile
+```
+
+#### Install tmc-check ####
+
+```
+$ sudo apt-get install check
+$ cd tmc-sandbox/uml/output/tmc-check
+$ rvmsudo make rubygems install clean
+```
+
+#### Run the test suite ####
+In the tmc-server root directory run
+```
+$ rvmsudo rake spec
+```
+If you are using Java 8 one of the tests will fail.
+
+#### Enable pghero stats ####
+Add to postgresql.conf following lines
+```
+shared_preload_libraries = 'pg_stat_statements'
+pg_stat_statements.track = all
+```
+
+### Background services ###
+
+After you get the test suite to pass, you can set up background services.
+
+1. Recheck your comet server config in `site.yml` and then do `rvmsudo rake comet:config:update`.
+2. Install init scripts: `rvmsudo rake comet:init:install`, `rvmsudo rake background_daemon:init:install`.
+3. Start the services: `sudo /etc/init.d/tmc-comet start`, `sudo /etc/init.d/tmc-background-daemon start`.
+4. If you use Apache, then make sure `public/` and `tmp/` are readable and install [mod_xsendfile](https://tn123.org/mod_xsendfile/). Configure XSendFilePath to the `tmp/cache` directory of the application.
+
+The application should not be deployed into a multithreaded server! It often changes the current working directory, which is a process-specific attribute. Each request should have its process all to itself. If you use Apache with, say, Passenger, then use the prefork MPM.
+
+### Startup ###
```

Comment:
This could be the development setup section and it could actually just suggest running `script/dev_env` and reading it if one wants to know more about what it actually starts.